src/mongo/db/ldap/ldap_manager_impl.cpp

```diff
@@ -160,12 +193,12 @@ class LDAPManagerImpl::ConnectionPoller : public BackgroundJob {
                 case ENOMEM: errname = "ENOMEM"; break;
                 }
                 LOG(2) << "poll() error name: " << errname;
-                //restart LDAP connection
+                //restart all LDAP connections... but why?
                 {
                     stdx::unique_lock<Latch> lock{_mutex};
-                    if(_poll_fd == fd.fd) {
-                        _poll_fd = -1;
-                        _manager->needReinit();
+                    if(!_poll_fds.empty()) {
+                        _poll_fds.clear();
+                        //_manager->needReinit();
```

Comment:
Can we replace this with:
if (!_poll_fds.empty()) {
    stdx::unique_lock<Latch> lock{_mutex};
    _poll_fds.clear();
}

Reply:
Oh, I see my proposal is wrong. Please ignore this.